### PR TITLE
integrate gregord and client CORE-3027

### PR DESCRIPTION
**Pull request by mmaxim**
@maxtaco I'm going to work on rigging up some tests, but you probably want to check out what I've done to make sure it looks reasonable.

Commit 61c4c45 has my new changes.

### CODE REVIEW

File: go/service/gregor.go [hunk 1]
```diff
@@ -96,6 +146,10 @@ func (g *gregorHandler) ShouldRetryOnConnect(err error) bool {
 func (g *gregorHandler) BroadcastMessage(ctx context.Context, m gregor1.Message) error {
 	g.G().Log.Debug("gregor handler: broadcast: %+v", m)
 
+	// Send message to local state machine
+	g.gregorCli.ConsumeMessage(m)
```
Comment (maxtaco) [inline go/service/gregor.go]: i wonder if you can dangle a boolean off of `context.Context` to say not to bother shipping back to the state machine if this comes from `sync`?

Comment (maxtaco) [inline go/service/gregor.go]: Maybe that's a bad idea though: https://github.com/golang/net/blob/master/context/pre_go17.go#L278-L279

**Comment by maxtaco** [comment]
In general looking good...

File: go/service/gregor.go (hunk 1 above)
Comment (mmaxim) [inline go/service/gregor.go]: Yeah that comment is oddly specific, I guess people wanted to do this using that function.

**Comment by mmaxim** [comment]
@maxtaco this is ready to look at again

File: go/service/gregor.go [hunk 2]
```diff
@@ -50,6 +129,44 @@ func (g *gregorHandler) HandlerName() string {
...
+
+	// Sync down everything from the server
+	if err = g.gregorCli.Sync(cli); err != nil {
+		g.G().Log.Errorf("gregor handler: error syncing from the server, bailing: %s", err)
+		return err
+	}
+
+	// Replay in-band messages
+	var msgs []gregor.InBandMessage
+	if msgs, err = g.gregorCli.InBandMessagesSince(t); err != nil {
```
Comment (mmaxim) [inline go/service/gregor.go]: Use the state machine directly since we just synced above.

File: go/service/gregor.go [hunk 3]
```diff
@@ -108,7 +238,8 @@ func (g *gregorHandler) BroadcastMessage(ctx context.Context, m gregor1.Message)
 		return g.handleOutOfBandMessage(ctx, obm)
 	}
 
-	return fmt.Errorf("gregor handler: both in-band and out-of-band message nil")
+	g.G().Log.Error("gregor handler: both in-band and out-of-band message nil")
+	return errors.New("invalid message")
```
Comment (mmaxim) [inline go/service/gregor.go]: Say gregor message

File: go/service/gregor.go [hunk 4]
```diff
@@ -98,6 +224,10 @@ func (g *gregorHandler) ShouldRetryOnConnect(err error) bool {
 func (g *gregorHandler) BroadcastMessage(ctx context.Context, m gregor1.Message) error {
 	g.G().Log.Debug("gregor handler: broadcast: %+v", m)
 
+	// Send message to local state machine
+	g.gregorCli.ConsumeMessage(m)
```
Comment (mmaxim) [inline go/service/gregor.go]: Expose state machine on g itself instead of this other way which is too confusing.

File: go/service/gregor.go [hunk 5]
```diff
@@ -359,20 +491,26 @@ func (g *gregorHandler) auth(ctx context.Context) error {
...
 	return nil
 }
 
-func (g *gregorHandler) connectNoTLS(uri *fmpURI) error {
+func (g *gregorHandler) connectNoTLS(uri *rpc.FMPURI) error {
+
 	g.G().Log.Debug("connecting to gregord without TLS")
 	t := newConnTransport(g.G(), uri.HostPort)
 	g.conn = rpc.NewConnectionWithTransport(g, t, keybase1.ErrorUnwrapper{}, true, keybase1.WrapError, g.G().Log, nil)
+	g.cli = g.conn.GetClient()
```
Comment (mmaxim) [inline go/service/gregor.go]: Add some comments about how this client should get used.

**Comment by maxtaco** [comment]
👍 After some function renaming and also some comments.  In another PR, let's have some tests.

**Pull request merged**